shared/p2p/hobbits/server_test.go

```diff
@@ -0,0 +1,243 @@
+package hobbits
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"reflect"
+	"testing"
+	"time"
+
+	ttl "github.com/ReneKroon/ttlcache"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/state"
+	"github.com/prysmaticlabs/prysm/beacon-chain/db"
+	"github.com/prysmaticlabs/prysm/shared/testutil"
+	"github.com/renaynay/go-hobbits/encoding"
+	"github.com/renaynay/go-hobbits/tcp"
+	"gopkg.in/mgo.v2/bson"
+
+	pb "github.com/prysmaticlabs/prysm/proto/beacon/p2p/v1"
+)
+
+func TestHobbitsNode_Hello(t *testing.T) {
+	fakeDB, err := db.NewDB("tmp/rene/")
+	if err != nil {
+		t.Errorf("could not generate new DB, %s", err.Error())
+	}
+
+	deposits, _ := testutil.SetupInitialDeposits(t, 5)
+	beaconState, err := state.GenesisBeaconState(deposits, 0, nil)
+
+	err = fakeDB.SaveState(context.Background(), beaconState)
+	if err != nil {
+		t.Errorf("error saving genesis state, %s", err.Error())
+	}
+
+	hobNode := Hobbits("127.0.0.1", 0, []string{}, fakeDB)
+
+	cache := ttl.NewCache()
+	cache.Set(string(make([]byte, 32)), true)
+
+	hobNode.MessageStore = cache
+
+	go func() {
+		hobNode.Listen()
+	}()
+
+	for {
+		if hobNode.Server.Addr() != nil {
+			break
+		}
+
+		time.Sleep(1)
```

Comment:
any way not to use sleep ? it makes tests brittle. Can you test this using channels instead